### PR TITLE
Implement conversion to and from the lumeo URL format

### PR DESCRIPTION
**Pull request by FSMaxB**

### CODE REVIEW

File: lumeo_api_client/src/files.rs [hunk 1]
```diff
@@ -53,6 +54,44 @@ pub struct File {
...
 
+impl File {
+    pub fn stream_url(&self) -> Url {
+        create_lumeo_file_url(self.id)
+    }
+}
+
+pub fn create_lumeo_file_url(file_id: Uuid) -> Url {
+    format!("lumeo://{file_id}").parse().unwrap_or_else(|error| {
+        unreachable!("Failed to parse generated Lumeo URL with error: {error}")
```
Comment (DmitrySamoylov) [inline lumeo_api_client/src/files.rs:65]: Doesn't `expect()` print the `error` value?

File: lumeo_api_client/src/files.rs [hunk 2]
```diff
@@ -53,6 +54,44 @@ pub struct File {
...
+
+#[derive(Debug, Error, PartialEq, Eq)]
+pub enum FileUrlError {
+    #[error("Invalid scheme, expected 'lumeo', got $0")]
+    InvalidScheme(String),
+    #[error("Missing host")]
+    MissingHost,
+    #[error("Invalid length")]
+    InvalidLength,
+    #[error("Invalid UUID: '$0'")]
```
Comment (DmitrySamoylov) [inline lumeo_api_client/src/files.rs:91]: I've never seen this syntax: `'$0'`, usually it's `{0}` ([example](https://docs.rs/thiserror/latest/thiserror/#example)).

**Review by DmitrySamoylov — APPROVED** [review]

File: lumeo_api_client/src/files.rs (hunk 1 above)
Comment (FSMaxB) [reply lumeo_api_client/src/files.rs:65]: Yes, it does, but I wanted to make it evident that this should never happen, so I used the `unreachable!` macro. But my paranoia kicked in that "if it happens even though it never should, we get the error message at least".

File: lumeo_api_client/src/files.rs (hunk 2 above)
Comment (FSMaxB) [reply lumeo_api_client/src/files.rs:91]: Good catch, yeah mine was just wrong!

File: lumeo_api_client/src/files.rs [hunk 3]
```diff
@@ -53,6 +54,44 @@ pub struct File {
...
+    if url.as_str().len() != ("lumeo://".len() + file_id.len()) {
+        return Err(FileUrlError::InvalidLength);
+    }
+
+    Uuid::parse_str(file_id).map_err(|_| FileUrlError::InvalidUuid(file_id.into()))
+}
+
+#[derive(Debug, Error, PartialEq, Eq)]
+pub enum FileUrlError {
+    #[error("Invalid scheme, expected 'lumeo', got $0")]
```
Comment (DmitrySamoylov) [inline lumeo_api_client/src/files.rs:85]: Same here.

Comment (FSMaxB) [reply lumeo_api_client/src/files.rs:85]: Oops, sorry.

**Review by DmitrySamoylov — APPROVED** [review]

**Pull request merged**